UnityProject/Assets/Scripts/MiniGames/MiniGameModule.cs

```diff
@@ -0,0 +1,31 @@
+using UnityEngine;
+
+namespace MiniGames
+{
+	/// <summary>
+	/// MiniGameModules are gameObjects that sit on the gameObject that has the IMiniGame Interface.
+	/// They are meant to be used when an object can house more than one minigame or if the minigame's behavior is too complex
+	/// to be stored in just the StartMiniGame() method.
+	/// </summary>
+	public abstract class MiniGameModule : MonoBehaviour
+	{
+		protected MiniGameResultTracker Tracker;
+		protected GameObject MiniGameParent;
+
+		/// <summary>
+		/// Use the Setup() function only when you're assigned events programatically and not from the inspector.
+		/// </summary>
+		public virtual void Setup(MiniGameResultTracker tracker, GameObject parent)
+		{
+			Tracker = tracker;
+			MiniGameParent = parent;
+			Tracker.OnStartGame.AddListener(StartMiniGame);
+		}
+
+		//Add your game setup/reset/ui show/etc here
+		public abstract void StartMiniGame();
+
+		//Add your logic here when a player has won or lost a game or has left the game
+		protected abstract void OnGameDone(bool t);
```

Comment:
Would it make sense to have this call `Tracker.OnGameEnd` so it's not forgotten?

Reply:
what do you mean

Reply:
I think he means giving the method a default implementation and call Tracker.OnGameEnd() inside this method

Reply:
I don't like the idea of turning this into a virtual void but since this PR has been taking forever.. eh, whatever. done.